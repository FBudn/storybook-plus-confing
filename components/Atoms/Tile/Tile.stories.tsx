import React from "react";
import { View } from "react-native";
import Tile from "./Tile";

const TileMeta = {
  title: "Atoms/Tile",
  component: Tile,
  args: {
    children: "Tile stories",
  },
  decorators: [
    (Story: any) => (
      <View
        style={{
          flex: 1,
          alignItems: "center",
          justifyContent: "center",
        }}
      >
        <Story />
      </View>
    ),
  ],
};

export default TileMeta;

export const Basic = {};
