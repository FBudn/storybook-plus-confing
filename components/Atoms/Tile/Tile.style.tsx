import { StyleSheet } from "react-native";

const TileStyles = StyleSheet.create({
  container: {
    flex: 0.7,
    backgroundColor: "white",
    borderRadius: 10,
    width: "80%",
    padding: 20,
    alignItems: "stretch",
    gap: 10,
  },
});

export default TileStyles;
