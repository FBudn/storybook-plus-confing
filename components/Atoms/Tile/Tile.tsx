import React from "react";
import { StyleSheet, View, Text } from "react-native";

export interface TileProps {
  children: any;
}

const Tile: React.FC<TileProps> = ({ children }) => {
  return <View style={styles.container}>{children}</View>;
};

const styles = StyleSheet.create({
  container: {
    flex: 0.7,
    backgroundColor: "white",
    borderRadius: 10,
    width: "80%",
    padding: 16,
    alignItems: "center",
  },
});

export default Tile;
