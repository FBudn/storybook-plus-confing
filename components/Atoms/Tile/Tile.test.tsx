import React from "react";
import { render, screen } from "@testing-library/react-native";
import TestRenderer from "react-test-renderer";
import Tile, { TileProps } from "./Tile";

const props: TileProps = {
  children: `Tile Test`,
  testId: `test-tile-id`,
};

test("Should render Tile component", () => {
  render(<Tile {...props} />);
  const TileElement = screen.getByTestId(`test-tile-id`);
  expect(TileElement).toBeOnTheScreen();
});

test("Should match snapshot:", () => {
  const TileElement = TestRenderer.create(<Tile {...props} />).toJSON();
  expect(TileElement).toMatchSnapshot();
});

test("Should render and have styles:", () => {
  render(<Tile {...props} />);
  const TileElement = screen.getByTestId(`test-tile-id`);
  expect(TileElement).toBeOnTheScreen();

  expect(TileElement).toHaveStyle({
    flex: 0.7,
    backgroundColor: "white",
    borderRadius: 10,
    width: "80%",
    padding: 16,
    alignItems: "stretch",
  });
});

test(`Should render with text`, () => {
  render(<Tile {...props} />);
  const TileElement = screen.getByTestId(`test-tile-id`);
  expect(TileElement).toBeOnTheScreen();
  expect(TileElement).toHaveTextContent(`Tile Test`);
});
