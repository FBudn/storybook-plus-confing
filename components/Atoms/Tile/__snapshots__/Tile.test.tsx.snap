// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot: 1`] = `
<View
  style={
    {
      "alignItems": "stretch",
      "backgroundColor": "white",
      "borderRadius": 10,
      "flex": 0.7,
      "gap": 10,
      "padding": 20,
      "width": "80%",
    }
  }
  testID="test-tile-id"
>
  Tile Test
</View>
`;
